mypy/checkpattern.py

```diff
@@ -244,6 +244,22 @@ def visit_sequence_pattern(self, o: SequencePattern) -> PatternType:
         # get inner types of original type
         #
         unpack_index = None
+
+        if isinstance(current_type, UnionType):
+            union_captures: dict[Expression, Type] = {}
+            union_items: list[Type] = []
+            for t in current_type.items:
+                typ, _, capture = self.accept(o, t)
+                if not isinstance(get_proper_type(typ), UninhabitedType):
+                    union_items.append(typ)
+                    union_captures.update(capture)
+            if len(union_items) == 0:
+                typ = UninhabitedType()
+            elif len(union_items) == 1:
```

Comment:
```suggestion
            num_items = len(union_items)
            if num_items == 0:
                typ = UninhabitedType()
            elif num_items == 1:
```
You can save a function call